aeron-tools/src/main/java/uk/co/real_logic/aeron/tools/MediaDriverOptions.java

```diff
@@ -0,0 +1,302 @@
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.GnuParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+import uk.co.real_logic.aeron.driver.Configuration;
+import uk.co.real_logic.agrona.concurrent.BackoffIdleStrategy;
+import uk.co.real_logic.agrona.concurrent.IdleStrategy;
+
+/**
+ * Created by bhorst on 4/10/15.
+ */
+public class MediaDriverOptions
+{
+    // Command line options for media driver
+    private final Options options;
+
+    // Properties containing Aeron settings.
+    private Properties properties;
+
+    // The 5 idle strategies available for the Media Driver
+    private IdleStrategy conductorIdleStrategy;
+    private IdleStrategy senderIdleStrategy;
+    private IdleStrategy receiverIdleStrategy;
+    private IdleStrategy sharedNetworkIdleStrategy;
+    private IdleStrategy sharedIdleStrategy;
+
+    public MediaDriverOptions()
+    {
+        options = new Options()
+                .addOption("p", "properties", true, "A properties file containing Aeron options.")
+                .addOption("h", "help", false, "Display help message.")
+                .addOption("c", "conductor", true, "IdleStrategy class for the conductor thread.")
+                .addOption("s", "sender", true, "IdleStrategy class for the sender thread.")
+                .addOption("r", "receiver", true, "IdleStrategy class for the receiver thread.")
+                .addOption("n", "network", true, "IdleStrategy class for the shared network thread.")
+                .addOption("a", "shared", true, "IdleStrategy class for the shared thread.");
+
+        // Idle strategies are null by default (Aeron should use its default if we pass null)
+        conductorIdleStrategy = null;
+        senderIdleStrategy = null;
+        receiverIdleStrategy = null;
+        sharedNetworkIdleStrategy = null;
+        sharedIdleStrategy = null;
+        properties = null;
+    }
+
+    /**
+     * Parse command line arguments for MediaDriverTool and call the setter functions.
+     * @param args Command line arguments.
+     * @return 0 on success, 1 if application should call {@link #printHelp(String)}
+     * @throws org.apache.commons.cli.ParseException On string parsing error.
+     */
+    public int parseArgs(final String[] args) throws ParseException
+    {
+        final CommandLineParser parser = new GnuParser();
+        final CommandLine command = parser.parse(options, args);
+
+        if (command.hasOption("help"))
+        {
+            // Inform application the user wants help message.
+            return 1;
+        }
+
+        if (command.hasOption("properties"))
+        {
+            this.properties = parseProperties(command.getOptionValue("properties"));
+        }
+        String defaultValue;
+        defaultValue = properties == null ? "null" : properties.getProperty("aeron.tools.mediadriver.sender");
```

Comment:
This feels like it should be simpler and cleaner. While I am sure it works, it is a bit hard to follow. And I can feel @mjpt777 twitching.